crypto/x509/x509_lu.c

```diff
@@ -681,9 +694,10 @@ int X509_STORE_CTX_get1_issuer(X509 **issuer, X509_STORE_CTX *ctx, X509 *x)
     if (ctx->check_issued(ctx, x, obj->data.x509)) {
         if (x509_check_cert_time(ctx, obj->data.x509, -1)) {
             *issuer = obj->data.x509;
-            X509_up_ref(*issuer);
+            if (!(ok = X509_up_ref(*issuer)))
+                *issuer = NULL;
             X509_OBJECT_free(obj);
-            return 1;
+            return ok;
```

Comment:
I think you want to return -1 on failure

Reply:
yes.